types/properties-reader/index.d.ts

```diff
@@ -0,0 +1,26 @@
+// Type definitions for properties-reader 0.0
+// Project: https://github.com/steveukx/properties
+// Definitions by: Zlatko Andonovski <https://github.com/Goldsmith42/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace PropertiesReader {
+    type Value = string|number|boolean;
+
+    interface Reader {
+        get(propertyName: string): Value|null;
+        getRaw(propertyName: string): string|null;
+        path(): Object;
+        append(path: string): Reader;
+        read(properties: string): Reader;
+        set(propertyName: string, value: Value): Reader;
+        length: number;
+        each(iterator: (key: string, value: Value) => void, scope?: Object): Reader;
```

Comment:
So, the value of `scope` is the value of `this` inside the iterator function -- that's what the second overload there is trying to say. And it sounds like the first overload should use `this: Reader`. Don't know why tslint is giving you problems there; you could add a `// tslint:disable-next-line unified-signatures`.